src/builder/server.js

```diff
@@ -192,12 +193,20 @@ export default (webpackConfig, options = {}) => {
 }
 
 export function getItemOrContinue(collection, req, res, next) {
-  const item = collection.find((item) => item.__url === req.originalUrl)
+  const __url = req.originalUrl
+    .replace(process.env.BASENAME, "/")
```

Comment:
Well, as you said, using env var will make this harder to test. I revert those change locally. Will push them right now